File: plugins/discourse-workflows/lib/discourse_workflows/node_data_shape.rb
# frozen_string_literal: true

module DiscourseWorkflows
  module NodeDataShape
    NODE_DIRECT_SETTING_KEYS = {
      "notes" => :notes,
      "notesInFlow" => :notes_in_flow,
      "alwaysOutputData" => :always_output_data,
      "onError" => :on_error,
      "continueOnFail" => :continue_on_fail,
    }.freeze

    FORM_TRIGGER_TYPE = "trigger:form"
    FORM_TRIGGER_WEBHOOK_ID_KEY = "webhookId"

    module_function

    def form_trigger?(node_type)
      node_type == FORM_TRIGGER_TYPE ||
        (node_type.respond_to?(:identifier) && node_type.identifier == FORM_TRIGGER_TYPE)
    end
  end
end

File: plugins/discourse-workflows/admin/assets/javascripts/admin/lib/workflows/node-data-shape.js
export const NODE_DIRECT_SETTING_KEY_ATTRIBUTES = Object.freeze({
  notes: "notes",
  notesInFlow: "notes_in_flow",
  alwaysOutputData: "always_output_data",
  onError: "on_error",
  continueOnFail: "continue_on_fail",
});
export const NODE_DIRECT_SETTING_KEYS = Object.freeze(
  Object.keys(NODE_DIRECT_SETTING_KEY_ATTRIBUTES)
);
export const FORM_TRIGGER_TYPE = "trigger:form";
export const FORM_TRIGGER_WEBHOOK_ID_KEY = "webhookId";

export function isFormTriggerNodeType(nodeType) {
  return nodeType === FORM_TRIGGER_TYPE;
}
